### DISCUSSION TITLE
Splitting the core.phel

### DISCUSSION THREAD

**Discussion by Chemaclass** [phel-lang/phel-lang#160] [Ideas]
What do you think about grouping and splitting the `core.phel` into multiple files?

I am not sure right now about how exactly, but I while reading the `core.phel` I realise that this grouping is done somehow by the comment delimiters such as:
```
# ----------------
# Print operations
# ----------------
```
These are all of them:
- Basic methods for quasiquote and destructure
- Basic macros
- Basic constructors
- Basic sequence operations
- Boolean operation
- Control structure macros
- Type operation
- Sequence operation
- Sequence functions
- Set operation
- Function operation
- More Sequence operation
- Bitwise operation
- Math operation
- Print operations
- Threading macros

There are 16 of them. It makes sense to think about splitting them, same as we already did for the `http.phel`, `html.phel` and `test.phel`. 

As an idea, maybe we could keep all basics stuff in the "core",  and try to group the rest on their own files?
I am not so keen on having 16 different files and that you need to require all of them everywhere because you will likely use every function... On the other hand, I like the idea that you will require the files that you really need, so you don't waste performance "requiring and parsing" stuff that you don't really need. And not only performance, but you will also group every function within their context, so the maintainability of each file will be easier in my opinion.

What do you think?

**Comment by jenshaase** [comment]
I will have a look if there is a way to split the files but keep them under the same namespace. Then we have both advantages.

**Reply by Chemaclass** [reply]
yes, I think that's possible and we can do it out of the box. Let's try this first and see how it was 👍🏼

**Comment by jenshaase** [comment]
I created an issue for that #169

**Comment by JesusValeraDev** [comment]
@Chemaclass @jenshaase can we close this?

**Reply by jenshaase** [reply]
This is a discussion. Can we close a discussion?

**Reply by Chemaclass** [reply]
I locked it... I don't find any "close" action for a discussion 😂

**Reply by JesusValeraDev** [reply]
No idea 😆

**Comment by Chemaclass** [comment]
This was done (https://github.com/phel-lang/phel-lang/issues/169) but undone later when working on the compile command (https://github.com/phel-lang/phel-lang/pull/293). So I will mark this as `wontdo`.
